src/utils/cleeng.ts

```diff
@@ -1,11 +1,29 @@
 import type { AccessModel } from '../../types/Config';
+import type { MediaOffer } from '../../types/media';
 
 export const isAllowedToWatch = (
   accessModel: AccessModel,
   isLoggedIn: boolean,
-  itemRequiresSubscription: boolean,
+  isItemFree: boolean,
   hasSubscription: boolean,
-): boolean =>
-  accessModel === 'AVOD' ||
-  (accessModel === 'AUTHVOD' && (isLoggedIn || !itemRequiresSubscription)) ||
-  (accessModel === 'SVOD' && (hasSubscription || !itemRequiresSubscription));
+  isEntitled?: boolean,
+): boolean => {
+  if (accessModel === 'AVOD') return true;
+  if (isItemFree) return true;
+  if (accessModel === 'AUTHVOD' && isLoggedIn) return true;
+  if (accessModel === 'SVOD' && hasSubscription) return true;
+  if (accessModel === 'SVOD' && isEntitled) return true;
+  if (accessModel === 'TVOD' && isEntitled) return true;
+
+  return false;
+};
+
+export const filterCleengMediaOffers = (offerIds: string = ''): MediaOffer[] => {
+  return offerIds
+    .split(',')
+    .reduce<MediaOffer[]>(
+      (offers, offerId) =>
+        offerId?.indexOf('cleeng:') === 0 ? [...offers, { offerId: offerId.slice(offerId.indexOf(':')), forced: offerId[0] === '!' }] : offers,
```

Comment:
Would be nice to add some comments explaining what this is doing

Reply:
Sure!